extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php

```diff
@@ -522,8 +526,9 @@ private function saveUsersAvatar( $userId = null, $data = null ) {
 			if ( empty( $wgAvatarsUseService ) ) {
 				// TODO: $user->getTouched() get be used to invalidate avatar URLs instead
 				$user->setGlobalAttribute( 'avatar_rev', date( 'U' ) );
-				$user->saveSettings();
 			}
+
+			$user->saveSettings();
```

Comment:
This actually causes a problem when saving a custom avatar.

On L514 we save an avatar via the avatar service, which writes the new avatar value directly to the attribute service, which then makes a call to MW to clear the user cache. But at no point is that new value for the avatar added to this user object and its instance variables.

So when we call  `saveSettings()` here we end up overwriting that new avatar value with whatever old value this user object had.